### PR TITLE
Pull secret change window

### LINKED ISSUE #41: Deal with pull-secret
* [x] send by API (ignore the file option)
  * [x] use API to determine if set
  
----

https://github.com/code-ready/tray-electron/blob/23c0e96faeb2d0ac9936652f64d420e513a7baf1/commander.js#L119-L128

https://github.com/code-ready/tray-electron/blob/23c0e96faeb2d0ac9936652f64d420e513a7baf1/commander.js#L130-L138

### LINKED ISSUE #61: Check for defined pull-secret before starting
When the preset is set to "OpenShift" the start action will check if the pullsecret has been defined.

If not, a pull secret change window will be show and allows the user to specify one and store it.

### LINKED ISSUE #75: Pull-secret input component for reuse in Wizard and Configuration change dialog
* [x] Create new component: `PullSecreInputCard` -> https://github.com/code-ready/crc-react-components/pull/13
  * [x] Change the Configuration to open a change dialog -> https://github.com/code-ready/crc-react-components/issues/15, waiting on merge of https://github.com/code-ready/tray-electron/pull/78
  * [x] the Change dialog will show a new component: [`PullSecretInput`](https://github.com/code-ready/crc-react-components/issues/12)
  * [x] this has to be re-used in wizard
  * [ ] and cockpit-crc (optional)

### PR DESCRIPTION
**Pull request by gbraad**
Fixes #75

image

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -588,7 +607,19 @@ ipcMain.on('config-load', async (event, args) => {
...
+// Pull secret specific
+// ------------------------------------------------------------------------- */
+
+ipcMain.on('open-pullsecret-window', async (event, args) => {
+  pullsecretChangeWindow.show();
+});
+
 ipcMain.on('pullsecret-change', async (event, args) => {
+
+    console.log(args.pullsecret);
```
Comment (anjannath) [inline main.js]: debugging left over?

Comment (gbraad) [reply main.js]: yep, removed in the pull secret startup check.

**Comment by anjannath** [comment]
One thing from the UI side that we can get to later is, the pull-secret change window is now a separate window that is not related to the config window visually  

i think we can use a modal for it, since it is a sub-window of config. specifically this component from patternfly: https://www.patternfly.org/v4/components/modal#with-form

**Comment by gbraad** [comment]
> i think we can use a modal for it, since it is a sub-window of config. specifically this component from patternfly:
> https://www.patternfly.org/v4/components/modal#with-form

This is what I wanted to do, but forgot ... It is more relevant for #61, but the current API is not reliable/working for this: https://github.com/code-ready/crc/issues/2950

**Comment by anjannath** [comment]
> > i think we can use a modal for it, since it is a sub-window of config. specifically this component from patternfly:
> > https://www.patternfly.org/v4/components/modal#with-form
> 
> This is what I wanted to do, but forgot ... It is more relevant for #61, but the current API is not reliable/working for this: [code-ready/crc#2950](https://github.com/code-ready/crc/issues/2950)

yeah, is not working for the 'podman' preset atm, i think @praveenkumar will look into this!

**Comment by gbraad** [comment]
> yeah, is not working for the 'podman' preset atm, i think @praveenkumar will look into this!

Can you check this again? It works as expected for me.
For me the issues were caused by segment's API being unreachable within the timeout.

**Comment by anjannath** [comment]
@gbraad i am not able quit the tray by clicking on the exit menu now, i think you need to also do `pullsecretChangeWindow.destroy()` in the `quitApp()`  function

**Comment by gbraad** [comment]
Added

**Review by anjannath — APPROVED** [review]

**Pull request merged**